### DISCUSSION TITLE
Automatic Performance Instrumentation Preview

### DISCUSSION THREAD

**Discussion by philipphofmann** [getsentry/sentry-cocoa#1187] [Show and tell]
With 7.1.0-beta.0, we released a preview of automatic performance instrumentation. Please [checkout the docs](https://docs.sentry.io/platforms/apple/performance/instrumentation/automatic-instrumentation/) for more info. We highly appreciate your feedback as a comment below.

**Comment by stsandro** [comment]
Does this work with SwiftUI apps? I'm setting up Sentry in the `init()` of my app struct as follows:

```swift
SentrySDK.start { options in
          // basic
          options.dsn = "https://1a2b3c4d5e6f7890@sentry.mydomain.com/3"
          options.debug = true
          options.diagnosticLevel = SentryLevel.warning
          options.enabled = true
          options.environment = "development"
          // advanced
          options.tracesSampleRate = 1.0 // Sampling 100% - In Production you probably want to adjust this
          options.enableAutoSessionTracking = true
          options.enableAutoPerformanceTracking = true
          options.enableUIViewControllerTracking = true
          options.enableNetworkTracking = true
          
          options.enableFileIOTracking = true
          options.enableProfiling = true
}
```

I don't see any tracking data comming in. I've tested with a custom transaction like:

```swift
let heavyOperation = SentrySDK.startTransaction()

// do something

heavyOperation.finsh()
``` 

That showed up in the performance dashboard, but nothing else.

**Reply by philipphofmann** [reply]
No, automatic instrumentation doesn't work yet with SwiftUI, see https://docs.sentry.io/platforms/apple/performance/instrumentation/automatic-instrumentation/.

We have an issue for this and plan to start working on this in the next weeks: https://github.com/getsentry/sentry-cocoa/issues/1737.

**Reply by stsandro** [reply]
Thanks for the heads up. 🙏 I'll monitor that issue

**Reply by armcknight** [reply]
Hi @stsandro , it looks like we did some work on that issue in #2271 that was released in 8.0.0, have you been able to take that update? I'm going to close this discussion as resolved, but feel free to reopen if you have any further feedback, or open a new issue. 🙏🏻

**Comment by papanton** [comment]
Is it possible for apps that use SwiftUI to still maintain the URLSession network tracking?

**Reply by philipphofmann** [reply]
We don't have support for SwiftUI auto-generated transactions yet. @brustolin, is working on that right now. And yes, it will be possible. Currently, we only attach network spans to auto-generated transactions. So if you use SwiftUI, you don't get auto-generated transactions, and hence it doesn't work right now.

**Reply by armcknight** [reply]
Hi @papanton , we did some work in https://github.com/getsentry/sentry-cocoa/pull/2271 that was released in 8.0.0, have you been able to take that update, and did it help with your use case? I'm going to close this discussion as resolved as it's been stale for a while, but feel free to reopen if you have any further feedback, or open a new issue. 🙏🏻
